src/Slider.test.tsx

```diff
@@ -144,7 +146,9 @@ describe('UpsellSlider', () => {
 
             await userEvent.click(screen.getByTestId('1'));
 
-            expect(clickSpy).toHaveBeenCalled();
+            await waitFor(() => {
+                expect(clickSpy).toHaveBeenCalled();
```

Comment:
I'd prefer to use `toHaveBeenCalledTimes()` over `toHaveBeenCalled()` (and when there are arguments in addition to `toHaveBeenCalledWith()`).

By using `toHaveBeenCalledTimes` instead of `toHaveBeenCalled` we can be sure we didn't introduce side effects by it being called more times than expected.

Reply:
Fair enough, however due the absence of measurements on offsetLeft/clientWidth, the arguments on the call don't make sense (it measures a left movement of 0) so that is why I omitted them.